src/api/db/data/20170306084550_remove_duplicate_repositories.rb

```diff
@@ -0,0 +1,50 @@
+class RemoveDuplicateRepositories < ActiveRecord::Migration[5.1]
+  def up
+    old = CONFIG['global_write_through']
+    CONFIG['global_write_through'] = false
+
+    Repository.transaction do
+      sql = <<-SQL
+        SELECT a.*
+        FROM repositories AS a
+        LEFT JOIN repositories AS b
+        ON a.db_project_id = b.db_project_id
+        AND a.name = b.name
+        WHERE a.id != b.id
+        AND a.remote_project_name IS NULL
+        AND b.remote_project_name IS NULL
+        GROUP BY a.db_project_id, a.name
+      SQL
+      repos_with_duplicates = Repository.find_by_sql(sql)
+
+      if repos_with_duplicates.any?
+        sql = <<-SQL
+          SELECT b.*
+          FROM repositories a
+          LEFT JOIN repositories AS b
+          ON a.db_project_id = b.db_project_id
+          AND a.name = b.name
+          WHERE a.id != b.id
+          AND a.remote_project_name IS NULL
+          AND b.remote_project_name IS NULL
+          AND a.id IN (#{repos_with_duplicates.map(&:id).join(', ')})
+        SQL
+        
+        duplicate_repos = Repository.find_by_sql(sql)
+        # We simply destroy the duplicate repository
```

Comment:
I think this 4 lines comment is not really clear to me. can you maybe rephrase it? And summarise it/short it would also be nice, I have the feeling that things like _simply_ and _does not make sense_ do not really add more information.

Reply:
There are also some typos: We simply destroy the **duplicated** **repositories**